### PR TITLE
fix various crashes-upon-exit

### PR DESCRIPTION
**Pull request by sethalves**
https://github.com/highfidelity/hifi/pull/13419 for rc69

https://highfidelity.fogbugz.com/f/cases/13685/MAC-Crash-report-displays-when-closing-interface-window

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10794/HighFidelity-Beta-PR13436-de43e55.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10794/HighFidelity-Beta-PR13436-de43e55.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10794/HighFidelity-Beta-Interface-PR13436-de43e55.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10794/HighFidelity-Beta-PR13436-de43e55.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10794/HighFidelity-Beta-Interface-PR13436-de43e55.exe

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122 crashes less on PR build than on Master

**Comment by qligmaloney** [comment]
FAIL @conklin94122 PR build crashed more times when closing via clicking the "X" button than the latest Master build (8470).
Logs attached.

Master Build (8470): 2/10
PR13436-de43e55: 3/10

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2125077/hifi-log.txt)
[hifi-log 2.txt](https://github.com/highfidelity/hifi/files/2125078/hifi-log.2.txt)
[hifi-log 3.txt](https://github.com/highfidelity/hifi/files/2125080/hifi-log.3.txt)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10831/HighFidelity-Beta-PR13436-b1c920a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10831/HighFidelity-Beta-PR13436-b1c920a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10831/HighFidelity-Beta-Interface-PR13436-b1c920a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10831/HighFidelity-Beta-PR13436-b1c920a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10831/HighFidelity-Beta-Interface-PR13436-b1c920a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10891/HighFidelity-Beta-PR13436-c74d9ca.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10891/HighFidelity-Beta-PR13436-c74d9ca.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10891/HighFidelity-Beta-Interface-PR13436-c74d9ca.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10891/HighFidelity-Beta-PR13436-c74d9ca.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10891/HighFidelity-Beta-Interface-PR13436-c74d9ca.exe

**Comment by qligmaloney** [comment]
Currently testing

**Comment by qligmaloney** [comment]
PASS @conklin94122 PR build crashed less than tip of Master build as shown below

File: interface/src/ui/ApplicationOverlay.cpp [hunk 1]
```diff
@@ -50,6 +50,8 @@ ApplicationOverlay::~ApplicationOverlay() {
         geometryCache->releaseID(_magnifierBorder);
         geometryCache->releaseID(_qmlGeometryId);
     }
+
+    DependencyManager::destroy<GeometryCache>();
```
Comment (jherico) [inline interface/src/ui/ApplicationOverlay.cpp]: this seems like a really weird place to put this.

Comment (AndrewMeadows) [reply interface/src/ui/ApplicationOverlay.cpp]: Maybe put it in `~Application()`?

Comment (sethalves) [reply interface/src/ui/ApplicationOverlay.cpp]: okay.  let's merge this one and I'll try to get this line back in Application.cpp again for the next one.  I have a couple more osx crash-on-exits to figure out.

**Pull request merged**